Error when Creating Database

I'm using MacOS High Sierra 10.13.1 and am able to build and install ComDB2.
I started the pmux service(`pmux -n`) and added my hostname in `/opt/bb/etc/cdb2/config/comdb2.d`. 
Now when I run the command to create the database: `comdb2 --create --dir ~/db testdb`, I'm getting the following error log.
```
resolve_dc(Saatviks-MBP.wv.cc.cmu.edu) open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.!
machine_class_default(Saatviks-MBP.wv.cc.cmu.edu) open rc -1 cdb2_get_dbhosts: no default_type entry in comdb2db config.!
ASYNCHRONOUS MODE
LOG DELETE ENABLED
LOG DELETE POLICY: delete all eligible log files
NEW MASTER NODE fakehost
[ERROR] Failed to open low level meta table, rc: 2
[FATAL] Failed to open low level meta table, rc: 0
[FATAL] failed to start
```

--- a/berkdb/txn/txn.c
+++ b/berkdb/txn/txn.c
@@ -1530,7 +1530,7 @@ __txn_discard(txnp, flags)
 		__os_free(dbenv, freep);
 	}
 
-	if ((ret = pthread_setspecific(txn_key, NULL)) != 0)
+	if (dbenv->tx_perfect_ckp && (ret = pthread_setspecific(txn_key, NULL)) != 0)
 		return (ret);
 
 	return (0);
@@ -1874,7 +1874,7 @@ __txn_end(txnp, is_commit)
 		(void)__txn_checkpoint(dbenv, 0, 0, DB_FORCE);
 	}
 
-	if ((ret = pthread_setspecific(txn_key, NULL)) != 0)
+	if (dbenv->tx_perfect_ckp && (ret = pthread_setspecific(txn_key, NULL)) != 0)
 		return (ret);
 
 	return (0);

--- a/berkdb/txn/txn_method.c
+++ b/berkdb/txn/txn_method.c
@@ -42,7 +42,7 @@ static int __txn_set_logical_start __P((DB_ENV *,
 static int __txn_set_logical_commit __P((DB_ENV *,
 	int (*)(DB_ENV *, void *, u_int64_t, DB_LSN *)));
 
-int gbl_use_perfect_ckp = 1;
+int gbl_use_perfect_ckp = 0;
 pthread_key_t txn_key;
 static pthread_once_t init_txn_key_once = PTHREAD_ONCE_INIT;
 

--- a/tests/leakcheck.test/lrl.options
+++ b/tests/leakcheck.test/lrl.options
@@ -2,3 +2,4 @@ table t1 t1.csc2
 setattr MIN_KEEP_LOGS 2
 setattr MIN_KEEP_LOGS_AGE 10
 setattr PRIVATE_BLKSEQ_MAXAGE 10
+perfect_ckp 1

--- a/tests/tunables.test/t00_all_tunables.expected
+++ b/tests/tunables.test/t00_all_tunables.expected
@@ -587,7 +587,7 @@
 (name='participantid_bits', description='Number of bits allocated for the participant stripe ID (remaining bits are used for the update ID).', type='INTEGER', value='0', read_only='N')
 (name='pause_moveto', description='pause_moveto', type='BOOLEAN', value='OFF', read_only='N')
 (name='penaltyincpercent', description='', type='INTEGER', value='20', read_only='Y')
-(name='perfect_ckp', description='', type='INTEGER', value='1', read_only='Y')
+(name='perfect_ckp', description='', type='INTEGER', value='0', read_only='Y')
 (name='pflt_readahead', description='Enable prefaulting of readahead operations', type='BOOLEAN', value='ON', read_only='N')
 (name='pflt_toblock_lcl', description='Prefault toblock operations locally', type='BOOLEAN', value='ON', read_only='N')
 (name='pflt_toblock_rep', description='Prefault toblock operations on replicants', type='BOOLEAN', value='ON', read_only='N')

